packages/core/src/libraries/protected-app.ts

```diff
@@ -58,8 +77,50 @@ export const createProtectedAppLibrary = (queries: Queries) => {
     );
   };
 
+  /**
+   * Build application data for protected app
+   * check if subdomain is valid
+   * generate host based on subdomain
+   * generate default protectedAppMetadata based on host and origin
+   * generate redirectUris and postLogoutRedirectUris based on host
+   */
+  const checkAndBuildProtectedAppData = async ({
+    subDomain,
+    origin,
+  }: {
+    subDomain: string;
+    origin: string;
+  }): Promise<Pick<Application, 'protectedAppMetadata' | 'oidcClientMetadata'>> => {
+    const error = new RequestError({
+      code: 'application.protected_application_subdomain_exists',
+      status: 422,
+    });
+    assertThat(isValidSubdomain(subDomain), error);
```

Comment:
The error message is not accurate here? inValidSubdomain !== protected_application_subdomain_exists